docs/conf.py

```diff
@@ -0,0 +1,314 @@
+        # The following module is only used /streams/developer-guide/
+        elif role == 'streams-apidocs-kstreambuilder':
+            linkPrefix = '../javadocs/org/apache/kafka/streams/kstream/KStreamBuilder.html'
+        # The following module is only used /streams/developer-guide/
+        elif role == 'streams-apidocs-streamsbuilder':
+            linkPrefix = '../javadocs/org/apache/kafka/streams/StreamsBuilder.html'
+        # The following module is only used /streams/developer-guide/
+        elif role == 'streams-apidocs-store-inmem':
+            linkPrefix = '../javadocs/org/apache/kafka/streams/state/Stores.InMemoryKeyValueFactory.html'
+        # The following module is only used /streams/developer-guide/
+        elif role == 'streams-apidocs-store-persistent':
+            linkPrefix = '../javadocs/org/apache/kafka/streams/state/Stores.PersistentKeyValueFactory.html'
+        # The following module is only used /streams/developer-guide/
+        elif role == 'streams-apidocs-timewindowedkstream':
+            linkPrefix = '../javadocs/org/apache/kafka/streams/kstream/TimeWindowedKStream.html'
+        # The following module is only used /streams/developer-guide/
+        elif role == 'streams-apidocs-sessionwindowedkstream':
+            linkPrefix = '../javadocs/org/apache/kafka/streams/kstream/SessionWindowedKStream.html'
+        linktext = tokens[0]
+        if len(tokens) == 2:
+            ref = linkPrefix + tokens[1]
+        else:
+            ref = linkPrefix + tokens[0]
+            linktext = ref
+        node = nodes.reference(rawtext, utils.unescape(linktext), refuri=ref, **options)
+        return [node], []
+    # add link macros
+    app.add_role('kafka-file', makeLink)
+    app.add_role('streams-apidocs-kstream', makeLink)
+    app.add_role('streams-apidocs-ktable', makeLink)
+    app.add_role('streams-apidocs-kgroupedstream', makeLink)
+    app.add_role('streams-apidocs-kgroupedtable', makeLink)
+    app.add_role('streams-apidocs-kstreambuilder', makeLink)
+    app.add_role('streams-apidocs-sessionwindowedkstream', makeLink)
+    app.add_role('streams-apidocs-streamsbuilder', makeLink)
+    app.add_role('streams-apidocs-store-inmem', makeLink)
+    app.add_role('streams-apidocs-store-persistent', makeLink)
+    app.add_role('streams-apidocs-timewindowedkstream', makeLink)
+
+# Add any paths that contain templates here, relative to this directory.
+templates_path = ['_templates']
+
+# The suffix of source filenames.
+source_suffix = '.rst'
+
+# The encoding of source files.
+#source_encoding = 'utf-8-sig'
+
+# The master toctree document.
+master_doc = 'index'
+
+# General information about the project.
+project = u'Apache Kafka'
+copyright = u'2018, Apache Software Foundation'
+
+# The version info for the project you're documenting, acts as replacement for
+# |version| and |release|, also used in various other places throughout the
```

Comment:
Yup, we should replace the {{fullVersion}} etc macros with this. Just to keep in mind that in AK we also have other macros so we should also add them here, for example:

```
major_version_prefix = '0.11'
doc_version = '0110'
```

Reply:
Full set I can find is per https://github.com/apache/kafka/blob/trunk/docs/js/templateData.js#L19-L24. Afaict, these are all addressed. This actually doesn't address the `doc_version` you mentioned, but updated PR includes this as `doc_url_version`, which is only currently used for javadocs links. I actually don't see anywhere something like `major_version_prefix` is used -- if there's a case you're aware of can you point me towards it?